app/src/main/java/io/homeassistant/companion/android/onboarding/integration/MobileAppIntegrationPresenterImpl.kt

```diff
@@ -0,0 +1,56 @@
+package io.homeassistant.companion.android.onboarding.integration
+
+import android.os.Build
+import android.util.Log
+import io.homeassistant.companion.android.BuildConfig
+import io.homeassistant.companion.android.domain.integration.DeviceRegistration
+import io.homeassistant.companion.android.domain.integration.IntegrationUseCase
+import kotlinx.coroutines.*
+import javax.inject.Inject
+
+class MobileAppIntegrationPresenterImpl @Inject constructor(
+    private val view: MobileAppIntegrationView,
+    private val integrationUseCase: IntegrationUseCase
+) : MobileAppIntegrationPresenter {
+
+    companion object {
+        private const val TAG = "IntegrationPresenter"
+    }
+
+    private val mainScope: CoroutineScope = CoroutineScope(Dispatchers.Main + Job())
+
+    override fun onRegistrationAttempt() {
+
+        view.showLoading()
+
+        mainScope.launch {
+            val deviceRegistration = DeviceRegistration(
+                BuildConfig.APPLICATION_ID,
+                "Home Assistant",
```

Comment:
@balloob is that ok for you or should we use something like `Home Assistant Android` as app_name ?

Reply:
It's maybe kind of a duplicate information as we already have the os name.